### PR TITLE
feat: migrate to flakes

### PR DESCRIPTION
**Pull request by ocind**

### CODE REVIEW

File: chart/Chart.tpl.yaml [hunk 1]
```diff
@@ -4,6 +4,6 @@ description: Sample Chart Description
 
 type: application
 
-version: {{.Env.VERSION}}
+version: { { .Env.VERSION } }
```
Comment (kirinnee) [inline chart/Chart.tpl.yaml]: this is wrong change
```suggestion
version: {{ .Env.VERSION }}
```

File: nix/pre-commit.nix [hunk 2]
```diff
@@ -0,0 +1,89 @@
+{ formatter, pre-commit-lib, packages }:
+pre-commit-lib.run {
+  src = ./.;
+
+  # hooks
+  hooks = {
+    # formatter
+    treefmt = {
+      enable = true;
+      excludes = [ ".*vendor.*" ".*scaffold/templates/.*" ".*chart/.*" "pnpm-lock.yaml" "Changelog.md" "CommitConventions.md" ".*schema.json" ".*gattai_custom_actions/.*" ];
```
Comment (kirinnee) [inline nix/pre-commit.nix]: can remove `pnpm-lock.yaml` `.*scaffold/templates/.*` etc

**Review by kirinnee — CHANGES_REQUESTED** [review]

File: nix/pre-commit.nix (hunk 2 above)
Comment (ocind) [reply nix/pre-commit.nix]: fixed

File: chart/Chart.tpl.yaml (hunk 1 above)
Comment (ocind) [reply chart/Chart.tpl.yaml]: fixed

**Review by kirinnee — APPROVED** [review]

**Pull request merged**